File: client-clojure/src/sirqul_io_t_platform/specs/predicted_location_response.clj
(ns sirqul-io-t-platform.specs.predicted-location-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def predicted-location-response-data
  {
   (ds/opt :ownerId) int?
   (ds/opt :onDate) string?
   (ds/opt :onDateMilli) int?
   (ds/opt :hour) int?
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :distance) float?
   (ds/opt :matches) int?
   (ds/opt :averageDuration) float?
   (ds/opt :weightedScore) float?
   })

(def predicted-location-response-spec
  (ds/spec
    {:name ::predicted-location-response
     :spec predicted-location-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/PredictedLocationResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PredictedLocationResponse._

case class PredictedLocationResponse (
  ownerId: Option[Long],
onDate: Option[String],
onDateMilli: Option[Long],
hour: Option[Integer],
latitude: Option[Double],
longitude: Option[Double],
distance: Option[Double],
matches: Option[Long],
averageDuration: Option[Double],
weightedScore: Option[Double])

object PredictedLocationResponse {
  import DateTimeCodecs._

  implicit val PredictedLocationResponseCodecJson: CodecJson[PredictedLocationResponse] = CodecJson.derive[PredictedLocationResponse]
  implicit val PredictedLocationResponseDecoder: EntityDecoder[PredictedLocationResponse] = jsonOf[PredictedLocationResponse]
  implicit val PredictedLocationResponseEncoder: EntityEncoder[PredictedLocationResponse] = jsonEncoderOf[PredictedLocationResponse]
}
